pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha2"
+)
+
+// ResolvedResultRefs represents all of the ResolvedResultRef for a pipeline task
+type ResolvedResultRefs []*ResolvedResultRef
+
+// ResolvedResultRef represents a result ref reference that has been fully resolved (value has been populated).
+// If the value is from a Result, then the ResultReference will be populated to point to the ResultReference
+// which resulted in the value
+type ResolvedResultRef struct {
+	Value           v1alpha2.ArrayOrString
+	ResultReference v1alpha2.ResultRef
+	FromTaskRun     string
+}
+
+type resultRefResolver struct {
+	resolvedTaskByName map[string]*ResolvedPipelineRunTask
+}
+
+func newResultRefResolver(pipelineRunState PipelineRunState) *resultRefResolver {
+	resolvePipelineRunTaskByName := make(map[string]*ResolvedPipelineRunTask)
+	for _, resolvedPipelineRunTask := range pipelineRunState {
+		resolvePipelineRunTaskByName[resolvedPipelineRunTask.PipelineTask.Name] = resolvedPipelineRunTask
```

Comment:
PipelineRunState has a `toMap()` func that performs this work.  I suggest making that method public and then reusing it throughout this file.  I also think we could then remove the `resultRefResolver` type completely and simply call `toMap` in the places where we need to look up by pipeline task name.

Reply:
Ok, I am removing resultRefResolver completely. Let's see if the resulting code looks better.